src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncMethodBuilderMemberCollection.cs

```diff
@@ -196,22 +204,27 @@ internal static bool TryCreate(SyntheticBoundNodeFactory F, MethodSymbol method,
                     resultType = typeMap.SubstituteType(resultType).Type;
                 }
                 returnType = returnType.ConstructedFrom.Construct(resultType);
-                NamedTypeSymbol builderType;
-                MethodSymbol createBuilderMethod;
-                PropertySymbol taskProperty;
-                bool customBuilder = returnType.IsCustomTaskType(out builderType, out createBuilderMethod);
-                if (!customBuilder)
-                {
-                    builderType = F.WellKnownType(WellKnownType.System_Runtime_CompilerServices_AsyncTaskMethodBuilder_T);
-                    Debug.Assert((object)builderType != null);
-                    builderType = builderType.Construct(resultType);
-                }
+                NamedTypeSymbol builderType = null;
```

Comment:
Removed in https://github.com/dotnet/roslyn/pull/13405/commits/f2cdba83c83d34ddac4875def41674d352fc5b7e#diff-995d1bf33206eef8c980dbd08b8601b0L207